src/DBFTPlugin/Consensus/ConsensusService.cs

```diff
@@ -72,6 +72,8 @@ private void OnPersistCompleted(Block block)
         {
             Log($"Persisted {nameof(Block)}: height={block.Index} hash={block.Hash} tx={block.Transactions.Length} nonce={block.Nonce}");
             knownHashes.Clear();
+            if (context.Block.Index == block.Index && context.CommitSent)
```

Comment:
Why does it need to be Committed?
Maybe it directly received the block.

Reply:
Only committed node saves context.